core/tee/tee_svc_cryp.c

```diff
@@ -2175,22 +2176,24 @@ tee_svc_obj_ed25519_sign(struct ed25519_keypair *key,
 {
 	size_t n;
 	size_t ctx_len;
-	uint8_t ctx[256] = {0};
-	uint8_t ph_flag = 0;
-	uint8_t cx_flag = 0;
+	uint8_t *ctx = NULL;
+	bool ph_flag = false;
+	bool cx_flag = false;
 
-	for (n = 0u; n < num_params; n++) {
+	for (n = 0; n < num_params; n++) {
 		switch (params[n].attributeID) {
 		case TEE_ATTR_EDDSA_PREHASH:
-			ph_flag = 1;
+			ph_flag = true;
 			break;
 
 		case TEE_ATTR_EDDSA_CTX:
-			cx_flag = 1;
+			cx_flag = true;
 			ctx_len = params[n].content.ref.length;
 			if (ctx_len > 255)
 				return TEE_ERROR_BAD_PARAMETERS;
-
+			ctx = mempool_calloc(mempool_default, 1, 256);
```

Comment:
This must be freed when we're done with it.

Reply:
fixed by [9d501f0](https://github.com/OP-TEE/optee_os/pull/5486/commits/9d501f03af51c0623d67826dd3c43cfcfc4a864d), along with other fixes that I've missed before.